Update windowmediaplayer.py

binhvtle opened the pull request.

vasily-v-ryabov commented on examples/windowmediaplayer.py:
```diff
@@ -83,6 +80,22 @@ def windows_media():
+    # Select and play a song
+    song_path = "#push your path"
+    window = app.Open
+    window.Wait('ready')
```
I'd recommend using PEP8 complaint names: `wait` instaed of `Wait` and `click` instead of `Click`. Old names will be deprecated in 0.6.5 and removed in 0.7.0.

binhvtle commented:
The version of windows used is Window 10. I have just updated for this
platform.

On Mon, Jul 30, 2018 at 3:06 PM Vasily Ryabov <notifications@github.com>
wrote:

> *@vasily-v-ryabov* commented on this pull request.
>
> Hi @binhvtle <https://github.com/binhvtle> thanks for your contribution.
> It generally looks good to me. Just need to update which Windows version is
> used (Windows 7 ?). Because Windows Media Player changed a lot on Windows
> 10 so that separate example is necessary for newer OS.
> ------------------------------
>
> In examples/windowmediaplayer.py
> <https://github.com/pywinauto/pywinauto/pull/527#discussion_r206044206>:
>
> > @@ -83,6 +80,22 @@ def windows_media():
>      time.sleep(.5)
>
>      app.ChooseColumsn.Cancel.click()
> +    time.sleep(.5)
> +
> +    # Open a media file
> +    app.WindowsMediaPlayer.menu_select("File->Open")
> +
> +    # Select and play a song
> +    song_path = "#push your path"
> +    window = app.Open
> +    window.Wait('ready')
>
> I'd recommend using PEP8 complaint names: wait instaed of Wait and click
> instead of Click. Old names will be deprecated in 0.6.5 and removed in
> 0.7.0.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pywinauto/pywinauto/pull/527#pullrequestreview-141426599>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AnEycuVz26SSMHPiq3x-WDO6wIUCyTWxks5uLr6bgaJpZM4VlTOB>
> .
>


-- 

______________________________________________________________

*Binh Le* – QA Engineer ►Vietnam office ► M (84) 932 512 535 ► Skype:
lebinhvhit

QASymphony, Inc. ► www.qasymphony.com

Facebook <https://www.facebook.com/kmscareers?ref=hl> | Twitter
<https://twitter.com/KMSCareers> | LinkedIn
<http://www.linkedin.com/groups/KMS-Technology-jobs-4392676?trk=my_groups-b-grp-v>

vasily-v-ryabov commented:
Ah, OK. I've figured out why it's not working on my Windows 10. The default settings of my Windows Media Player doesn't show main menu. I had to enable showing main menu by toolbar (see the screenshot below).
![2018-07-30_11-32-06](https://user-images.githubusercontent.com/7419086/43394267-c1d7f8c4-9402-11e8-911d-aa4cb53578c8.png)

vasily-v-ryabov commented on examples/windowmediaplayer.py:
```diff
@@ -52,26 +51,24 @@ def windows_media():
 
-    app.WindowsMediaPlayer.menu_select("View->GoTo->Library")
+    # Wait for Window media player displays
+    time.sleep(3)
```
Such a hard-coded `time.sleep()` calls make the script a bit slow. It's better to call `app.WindowsMediaPlayer.SomeControlAppearedAfterPreviousAction.wait('ready')` (default timeout is up to 5 seconds). I will think if it's possible to do something inside `menu_select`.

vasily-v-ryabov commented on examples/windowmediaplayer.py:
```diff
@@ -83,6 +80,22 @@ def windows_media():
+    app.WindowsMediaPlayer.menu_select("File->Open")
+
+    # Select and play a song
+    song_path = "#push your path"
```
I'd suggest to use some standard sound from folder `C:\Windows\media` to make the script working out of the box.

vasily-v-ryabov commented on examples/windowmediaplayer.py:
```diff
@@ -83,6 +80,22 @@ def windows_media():
+    button.Click()
+
+    # Let the music plays
+    time.sleep(20)
```
This `time.sleep` is OK, of course. 👍 Maybe just need to align it with chosen track length.
